aion_pool/src/Miningcore/Blockchain/Aion/AionRewardsCalculator.cs

```diff
@@ -27,5 +41,13 @@ class AionRewardsCalculator
                 return (decimal) blockReward / magnitude;
             }
         }
+
+        public decimal calculateRewardWithTimeSpan(long timeSpan) {
+            if (timeSpan == 0) {
+                return 0;
```

Comment:
Is this true in the constant case? I feel that block reward would still be 4.5. So no reason to zero it out for timeSpan == 0.

Reply:
This is more an invalid block altogether I guess (so it should never happen)

Reply:
Yes, I think the case should not happens. But if it happen, I return a "0" in case the following out of index error.

Reply:
I think if we don't expect this to be possible, we should throw/assert

Reply:
updated the code, use throw to instead of return 0